lib/model/model.go

```diff
@@ -1539,13 +1540,121 @@ func sendIndexTo(minSequence int64, conn protocol.Connection, folder string, fs
 }
 
 func (m *Model) updateLocalsFromScanning(folder string, fs []protocol.FileInfo) {
-	m.updateLocals(folder, fs)
+	m.fmut.RLock()
+	folderCfg := m.folderCfgs[folder]
+	m.fmut.RUnlock()
+
+	folderrunner := m.folderRunners[folder]
+	var newfs []protocol.FileInfo
+
+	// if this folder is a "write/pull only" folder, then we'll have to undo any local changes
+	if folderCfg.Type == config.FolderTypeWriteOnly {
+		// reject and undo all local changes
+		// GAP: need to handle file renaming also (see rwfolder line 511+ comparing blocks)
+
+		fileDeletions := []protocol.FileInfo{}
+		dirDeletions := []protocol.FileInfo{}
+
+		for _, file := range fs {
+			if strings.Contains(file.Name, ".sync-conflict-") {
+				// this is a conflict copy, let's move on to the next file
+				continue
```

Comment:
Not handling this case caused issues, since it detects the sync conflict file as a new local file and once again wants to reject the local change, creating a new sync conflict file, ...